portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,38 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private Comment comments = new Comment();
+  private boolean isEmptyComment;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.setContentType("application/json");
+    String json = new Gson().toJson(comments);
+    response.getWriter().println(json);
+  }
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String comment = getComment(request);
+      if(isEmptyComment){
```

Comment:
Instead of making this a class level variable, I think it makes more sense just to inline this - so here, instead of checking `isEmptyComment`, just check `comment.isEmpty()` directly.

The reason here is scope. Adding the `isEmptyComment` variable, the scope of it is the entire class. So every method would be able to access it. Instead, having it inline means it's only accessible in the scope of this method. Since you probably don't need to check this boolean anywhere else, it makes sense for the scope to be just this method. Usually, you want to set the scope for data at the smallest level so it's easier to manage.